Apps/Src/Voltage.c

```diff
@@ -22,57 +27,73 @@ static uint32_t openWires[TOTAL_VOLT_WIRES];
 /** Voltage_Init
  * Initializes all device drivers including LTC6811 and GPIO to begin Voltage Monitoring
  * @param boards LTC6811 data structure that contains the values of each register
+ * @param voltageMutex pointer to mutex, meant to pass pointer to VoltageBuffer_Mutex
  * @return SUCCESS or ERROR
  */
-ErrorStatus Voltage_Init(cell_asic *boards){
+void Voltage_Init(cell_asic *boards){
 	// Record pointer
 	Minions = boards;
-
-	int8_t error = 0;
-	
+	//initialize mutex
+	OS_ERR err;
+	OSMutexCreate(&Voltage_Mutex,
+				  "Voltage Buffer Mutex",
+				  &err
+				);
+					
 	wakeup_sleep(NUM_MINIONS);
 	LTC6811_Init(Minions);
 	
+	//take control of mutex
+  	OSMutexPend(&MinionsASIC_Mutex, 0, OS_OPT_PEND_BLOCKING, NULL, &err);
+  	assertOSError(err);
 	// Write Configuration Register
 	LTC6811_wrcfg(NUM_MINIONS, Minions);
+	//release mutex
+  	OSMutexPost(&MinionsASIC_Mutex, OS_OPT_POST_NONE, &err);
+  	assertOSError(err);
 
 	// Read Configuration Register
+	// take control of mutex
+  	OSMutexPend(&MinionsASIC_Mutex, 0, OS_OPT_PEND_BLOCKING, NULL, &err);
+  	assertOSError(err);
 	wakeup_sleep(NUM_MINIONS);
-	error = LTC6811_rdcfg(NUM_MINIONS, Minions);
+	LTC6811_rdcfg_safe(NUM_MINIONS, Minions);
+	// release mutex
+  	OSMutexPost(&MinionsASIC_Mutex, OS_OPT_POST_NONE, &err);
+  	assertOSError(err);
 	
-	if(error == 0){
-		return SUCCESS;
-	}else{
-		return ERROR;
-	}
 }
 
 /** Voltage_UpdateMeasurements
  * Stores and updates the new measurements received
  * @param pointer to new voltage measurements
- * @return SUCCESS or ERROR
  */
-ErrorStatus Voltage_UpdateMeasurements(void){
-	int8_t error = 0;
-	
+void Voltage_UpdateMeasurements(void){
+	CPU_TS ts;
 	// Start Cell ADC Measurements
 	wakeup_idle(NUM_MINIONS);
 	LTC6811_adcv(ADC_CONVERSION_MODE,ADC_DCP,CELL_CH_TO_CONVERT);
```

Comment:
Do we need to hold a mutex to start ADC conversions and the like?

Reply:
I think so